tests/unittests/GraphOptzTest.cpp

```diff
@@ -386,6 +386,152 @@ TEST_F(GraphOptz, optimizeBatchNormAfterConvWithPred) {
+  auto *beta = mod_.createConstant(ElemKind::FloatTy, {4}, "beta");
+  auto *gamma = mod_.createConstant(ElemKind::FloatTy, {4}, "gamma");
+
+  Node *subC = mod_.createConstant(ElemKind::FloatTy, {3, 2, 2, 4}, "subC");
+  Node *mulC = mod_.createConstant(ElemKind::FloatTy, {3, 2, 2, 4}, "mulC");
+  Node *addC = mod_.createConstant(ElemKind::FloatTy, {3, 2, 2, 4}, "addC");
+  Node *divC = mod_.createConstant(ElemKind::FloatTy, {3, 2, 2, 4}, "divC");
+
+  // Fill tensors to check boundary values after the transformation.
+  std::vector<float> betaV = {1., 2., 3., 7.};
+  std::vector<float> gammaV = {4., 5., 6., 7.};
+
+  var->getPayloadMutable().getHandle<float>() = {1., 1., 1., 1.};
+  mean->getPayloadMutable().getHandle<float>() = {0., 0., 0., 0.};
+  beta->getPayloadMutable().getHandle<float>() = betaV;
+  gamma->getPayloadMutable().getHandle<float>() = gammaV;
+
+  // For at least one node (sub) make values within channel different, to test
+  // folding better.
+  const std::vector<float> subV = {1, 2., 3., 4.};
+  const float mulV = 4., addV = 3., divV = 2.;
+  auto subH = llvm::cast<Constant>(subC)->getHandle<float>();
+  subH = {1., 2., 3., 4., 1., 2., 3., 4., 1., 2., 3., 4., 1., 2., 3., 4.,
+          1., 2., 3., 4., 1., 2., 3., 4., 1., 2., 3., 4., 1., 2., 3., 4.,
+          1., 2., 3., 4., 1., 2., 3., 4., 1., 2., 3., 4., 1., 2., 3., 4.};
+
+  llvm::cast<Constant>(mulC)->getHandle<float>().clear(mulV);
+  llvm::cast<Constant>(addC)->getHandle<float>().clear(addV);
+  llvm::cast<Constant>(divC)->getHandle<float>().clear(divV);
+
+  BatchNormalizationNode *bn =
+      F_->createBatchNormalization("batch", input, beta, gamma, mean, var, 3);
+
+  auto *sub = F_->createSub("sub", bn, subC);
+  auto *mul = F_->createMul("mul", sub, mulC);
+  auto *add = F_->createAdd("add", addC, mul);
+  auto *div = F_->createDiv("div", divC, add);
+  auto *result = F_->createSave("save", div);
+
+  // Compile.
+  EXPECT_EQ(F_->getNodes().size(), 6);
+  ::glow::optimize(F_, CompilationMode::Infer);
+  EXPECT_EQ(F_->getNodes().size(), 2);
+
+  Constant *cs, *cb;
+
+  auto *newBn = llvm::dyn_cast<BatchNormalizationNode>(result->getInput());
+  ASSERT_TRUE(newBn);
+
+  cs = llvm::dyn_cast<Constant>(newBn->getScale());
+  cb = llvm::dyn_cast<Constant>(newBn->getBias());
+  ASSERT_TRUE(cs);
+  ASSERT_TRUE(cb);
+  ASSERT_TRUE(cs->getType()->isFPType());
+  ASSERT_TRUE(cb->getType()->isFPType());
+
+  auto hs = cs->getHandle<float>();
+  auto hb = cb->getHandle<float>();
+
+  // Verify that scale and offset are computed correctly.
+  for (size_t i = 0; i < 4; i++) {
+    const float expectedScale = gammaV[i] * mulV / divV;
+    const float expectedBias = ((betaV[i] - subV[i]) * mulV + addV) / divV;
+    EXPECT_EQ(expectedScale, hs.raw(i));
+    EXPECT_EQ(expectedBias, hb.raw(i));
+  }
+}
+
+/// Testing merge of single-user arithmetic operation chain (Sub, Mul, Add)
+/// into a BatchNorm.
+TEST_F(GraphOptz, FoldArithmeticChainAfterConvIntoBatchNorm) {
```

Comment:
Can you use `checkNumericalEquivalence()` in the test?

Reply:
check added